R/geom_params.R

```diff
@@ -0,0 +1,187 @@
+#' Does a layer use one of more specific parameters?
+#'
+#' \code{uses_geom_params} checks that a plot's geom layer uses a specific parameter.
+#'
+#' To specify a specific geom layer, either specify using position using the \code{i} index or
+#' by using a combination of \code{geom} function suffix name and \code{i} to check the ith layer that
+#' uses the geom.
+#'
+#' The \code{params} argument accepts a list that contains geom, stat, or aes
+#' parameters. This offers flexibility in certain situations where setting a
+#' parameter on a \code{geom_} function is actually setting a stat parameter or
+#' aes parameter. For example, in \code{geom_histogram(binwidth = 500)}, the
+#' \code{binwidth} is a stat parameter, while in
+#' \code{geom_histogram(fill = "blue")}, the \code{fill} is an aes parameter.
+#' \code{uses_geom_params} will take this into account and check geom, stat, and
+#' aes parameters.
+#'
+#' Note that `uses_geom_params()` can detect aes _parameters_, but not aes
+#' _mappings_. Parameters are set directly within a layer (e.g.
+#' `geom_point(color = "blue")`), while mappings are set using
+#' [`aes()`][ggplot2::aes] (e.g. `geom_point(aes(color = class))`).
+#'
+#' @param p A ggplot object
+#' @param geom A character string found in the suffix of a ggplot2 geom function,
+#'  e.g. \code{"point"}.
+#' @param ... <[`dynamic-dots`][rlang::dyn-dots]>
+#'   Named values or [character] strings.
+#'   Unnamed arguments will check whether any value was set for that parameter.
+#'   Named arguments will check whether the parameter with the same name has a
+#'   matching value.
+#'   Each argument should have a name matching a [ggplot][ggplot2::ggplot]
+#'   layer parameter.
+#'   Values may be passed as arguments or as list elements.
+#' @param params A named list of geom or stat parameter values, e.g.
+#'   \code{list(outlier.alpha = 0.01)}.
+#'   This list is combined with any inputs to `...`
+#' @param i A numerical index, e.g. \code{1}.
+#'
+#' @return A named logical vector of the same length as the number of inputs
+#'   to `...`.
+#' @family functions for checking geom parameters
+#' @export
+#'
+#' @examples
```

Comment:
Can we move `@examples` above the `@param`s please?

Reply:
Resolved in 3cf709f